viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    movie = {}
+    movie_set = {title, genre, rating}
+
+    if None in movie_set:
+        return None
+    else:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title != movie["title"]:
+            continue
+        else:
+            movie_index = user_data["watchlist"].index(movie)
+            removed_movie = user_data["watchlist"].pop(movie_index)
+            add_to_watched(user_data, removed_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    total = 0
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    
+    average = total / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+    
+    compare_dict = {}
+
+    if not user_data["watched"]:
+        return None
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre not in compare_dict:
+            compare_dict[genre] = 0
+        compare_dict[genre] += 1
+    
+    most_watched_genre = max(compare_dict, key=compare_dict.get)
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    friends_movie_list = []
+
+    for friend in user_data["friends"]:
+        friends_movie_list.extend(friend["watched"])
+    
+    unique_movies = []
+
+    for movie in user_data["watched"]:
+        if movie not in friends_movie_list:

```

Comment:
Using `in` on a list conceals a loop within it. It would be really nice if we could use a data structure like `set` instead, which has a very efficient `in` check. Unfortunately, we can't store an entire movie in a set, since we are using a dictionary to represent them, but what if we used a movie's title (an immutable string) to represent the entire movie? How could we build such a set, and how would we use it for this uniqueness check?